tests/integration/high_availability/test_upgrade.py

```diff
@@ -17,29 +30,50 @@
 
 TIMEOUT = 15 * 60
 
+MYSQL_APP_NAME = "mysql-k8s"
+TEST_APP_NAME = "test-app"
+
 
 @pytest.mark.group(1)
-async def test_build_and_deploy(ops_test: OpsTest) -> None:
+@pytest.mark.abort_on_fail
+async def test_deploy_latest(ops_test: OpsTest) -> None:
     """Simple test to ensure that the mysql and application charms get deployed."""
-    await high_availability_test_setup(ops_test)
+    await asyncio.gather(
+        ops_test.model.deploy(
+            MYSQL_APP_NAME,
+            application_name=MYSQL_APP_NAME,
+            num_units=3,
+            channel="8.0/edge",
+            trust=True,
+            config={"profile": "testing"},
+        ),
+        ops_test.model.deploy(
+            f"mysql-{TEST_APP_NAME}",
```

Comment:
this is changed due to helper [`get_application_name`](https://github.com/canonical/mysql-k8s-operator/blob/d73010c9eba7641a80b6b91982d468eaaba91a47/tests/integration/high_availability/high_availability_helpers.py#L82) matching the mysql-test-app occasionally (which I believe happens due to the deployment order, since I'm deploying both apps in parallel).
When it happens, we get errors like the `ensure_continous_writes` helper running actions against `mysql` instead of `mysql-test-app`.
I've changed this here, because is the only test that we do not build and deploy local mysql before all other actions.

Reply:
Please fire an separate ticket to optimise this. This CI/CD is important to merge and run for new PRs, the optimisation is logical but requires time for investigation.